docs/src/main/asciidoc/security-jpa-concept.adoc

```diff
@@ -0,0 +1,120 @@
+[id="security-jpa-concept"]
+= Quarkus Security with JPA
+include::_attributes.adoc[]
+:categories: security,jpa
+
+JPA is an identity provider, similar to xref:[JDBC], suitable for use with the xref:security-basic-authentication-concept.adoc[Basic] and xref:security-authentication-mechanisms-concept.adoc#form-auth[Form-based] Quarkus security mechanisms, which require a combination of username and password credentials.
+
+The JPA `IdentityProvider` creates a `SecurityIdentity` instance, which is used during user authentication to verify and authorize access requests making your Quarkus application secure.
+
+For an example of practical use of Basic authentication and JPA, see the xref:security-basic-authentication-tutorial.adoc[Secure a Quarkus application with Basic authentication and JPA] tutorial.
+
+
+== JPA entity specification
+
+Quarkus security offers a JPA integration to collect usernames, passwords, and roles, and store them into JPA database entities.
+
+The following JPA entity specification demonstrates how users' information needs to be stored in a JPA entity and properly mapped so that Quarkus can retrieve this information from a database.
+
+
+* The `@UserDefinition` annotation must be present on a JPA entity, no matter if link:https://quarkus.io/guides/hibernate-orm-panache[simplified Hibernate ORM with Panache] is used or not.
+
+* The `@Username` and `@Password` field types are always `String`.
+
+* The `@Roles` field must either be `String`, `Collection<String>`, or a `Collection<X>`, where `X` is an entity class with a single `String` field annotated as `@RolesValue`.
+
+* Each `String` role element type is parsed as a comma-separated list of roles.
+
```

Comment:
Please copy the section from the Basic Auth and JPA tutorial over here

Reply:
It will cause some duplication but but the Basic Auth and JPA tutorial is about getting started fast with Quarkus sec with the help of Basic Auth and JPA, while the purpose of this doc is to offer a complete JPA concept

Reply:
OK @sberyozkin . Adding now. Will rebase and squash.

Reply:
Done and done. See below from row 28.